plugin/action/throttle/throttle.go

```diff
@@ -103,40 +168,140 @@ func factory() (pipeline.AnyPlugin, pipeline.AnyConfig) {
 	return &Plugin{}, &Config{}
 }
 
+func (p *Plugin) syncWorker(ctx context.Context, jobCh <-chan limiter, wg *sync.WaitGroup) {
+	for {
+		select {
+		case <-ctx.Done():
+			return
+		case job := <-jobCh:
+			job.sync()
+			wg.Done()
+		}
+	}
+}
+
 func (p *Plugin) Start(config pipeline.AnyConfig, params *pipeline.ActionPluginParams) {
 	p.config = config.(*Config)
+	p.logger = params.Logger
 	p.pipeline = params.PipelineName
-	p.limiterBuff = make([]byte, 0)
+	p.limiterBuf = make([]byte, 0)
+	ctx, cancel := context.WithCancel(context.Background())
+	p.ctx = ctx
+	p.cancel = cancel
 
 	limitersMu.Lock()
-	limiters[p.pipeline] = map[string]*limiter{}
+	limiters[p.pipeline] = map[string]limiter{}
 	limitersMu.Unlock()
 
-	for _, r := range p.config.Rules {
-		p.rules = append(p.rules, NewRule(r.Conditions, complexLimit{r.Limit, r.LimitKind}))
+	format, err := pipeline.ParseFormatName(p.config.TimeFieldFormat)
+	if err != nil {
+		format = p.format
 	}
+	p.format = format
 
-	p.rules = append(p.rules, NewRule(map[string]string{}, complexLimit{p.config.DefaultLimit, p.config.LimitKind}))
+	if p.config.LimiterBackend == redisBackend {
+		if p.config.RedisBackendCfg.WorkerCount < 1 {
+			p.logger.Fatalf("workers_count must be > 0, passed: %d", p.config.RedisBackendCfg.WorkerCount)
+		}
+
+		p.redisClient = redis.NewClient(
+			&redis.Options{
+				Network:      "tcp",
+				Addr:         p.config.RedisBackendCfg.Endpoint,
+				Password:     p.config.RedisBackendCfg.Password,
+				ReadTimeout:  p.config.RedisBackendCfg.Timeout_,
+				WriteTimeout: p.config.RedisBackendCfg.Timeout_,
+			})
+
+		if pingResp := p.redisClient.Ping(); pingResp.Err() != nil {
+			p.logger.Fatalf("can't ping redis: %s", pingResp.Err())
+		}
+
+		ticker := time.NewTicker(p.config.RedisBackendCfg.SyncInterval_)
+
+		limitersMu.Lock()
+		wg := sync.WaitGroup{}
+		// run syncer at single throttle for one pipeline
+		if _, ok := limiterSyncMap[p.pipeline]; !ok {
```

Comment:
make this code more readable:
1. extract condition contents to separate function, 
2. don't use `else`
3. you don't need sync.Map as it already under mutex
4. use map[string]bool — we don't need to optimize memory here
5. what does `run syncer at single throttle for one pipeline` mean? `run sync only once per pipeline`?

e.g
```
limitersMu.Lock()
if !limiterSyncMap[p.pipeline] {
    limiterSyncMap[p.pipeline] = true
    runSync(p.pipeline)
}
limitersMu.Unlock()
```

Reply:
About point num 3: it's not sync map, just named sync to show it's purpose. I will rename it to avoid ambiguity

Reply:
Point 5: Correct, I meant `run sync only once per pipeline`, I renamed with this suggestion.

Reply:
All done except point 4: map[string]struct{} suits better then map[string]bool